Sources/Apodini/Semantic Model Builder/InterfaceExporter.swift

```diff
@@ -7,7 +7,13 @@ import Vapor
 protocol InterfaceExporter: RequestInjectableDecoder {
     init(_ app: Application)
 
-    func export(_ node: EndpointsTreeNode)
+    func export(_ endpoint: Endpoint)
+
+    func finishedExporting(_ root: EndpointsTreeNode)
```

Comment:
Will there be a real use-case for this function?
From my gRPC POV I can't imagine one (except logging "I'm done exporting" 😁)

Reply:
Yeah felt a bit dirty to me as well 😅 The use case for REST is that, in the case that the user didn't register a handler under the `/` path (not sure if they should actually), the REST exporter should automatically create a route handler there to be able to respond with links to all 'entry points'.

like the example:
```swift
var content: some Component {
  Group("user") {
    // ...
  }
  Group("articles") {
    // ...
  }
}
```

In order to be 100% HATEOS "compliant" a client should be able to do a request to `GET /` and be able to get links to `/user` and `/articles`. The REST exporter basically needs the information, are there any endpoints under the root path.
By the time of writing, I could see that we could remove the `node` argument, and thus fully hide `EndpointsTreeNode`s from the exporters. the `export` calls could just save the information of the presence of a `/` endpoint. But in any case there would need to be a `finishedExporting`. But the method is optional, as it provides a empty default implementations as I'm aware, that not many exporters need that.

Reply:
Nevermind, ich brauch ja auch die Info, wohin die root node alles linked, also relationship information. Die müsste ich irgendwie anders noch bekommen können um das zu entfernen.

Reply:
@moritzschuell Would you mind having a look at fe34827 which tries to completely hide the EndpointsTree to the exporters by creating a WebService model to store said information.

Reply:
Looks good to me 🙂